Extensions/Effects/JsExtension.js

```diff
@@ -27,127 +27,151 @@ module.exports = {
+      new gd.PropertyDescriptor(/* defaultValue= */ "0.5")
+        .setLabel(_("Intensity (between 0 and 1)"))
+        .setType("number")
     );
     nightProperties.set(
-      'opacity',
-      new gd.PropertyDescriptor(/* defaultValue= */ '0.5')
-        .setLabel(_('Opacity (between 0 and 1)'))
-        .setType('number')
+      "opacity",
+      new gd.PropertyDescriptor(/* defaultValue= */ "0.5")
+        .setLabel(_("Opacity (between 0 and 1)"))
+        .setType("number")
     );
 
     const noiseEffect = extension
-      .addEffect('Noise')
-      .setFullName(_('Noise'))
-      .setDescription(_('Add some noise on the rendered image'))
-      .addIncludeFile('Extensions/Effects/noise-pixi-filter.js');
+      .addEffect("Noise")
+      .setFullName(_("Noise"))
+      .setDescription(_("Add some noise on the rendered image"))
+      .addIncludeFile("Extensions/Effects/noise-pixi-filter.js");
     const noiseProperties = noiseEffect.getProperties();
     noiseProperties.set(
-      'noise',
-      new gd.PropertyDescriptor(/* defaultValue= */ '0.5')
-        .setLabel(_('Noise intensity (between 0 and 1)'))
-        .setType('number')
+      "noise",
+      new gd.PropertyDescriptor(/* defaultValue= */ "0.5")
+        .setLabel(_("Noise intensity (between 0 and 1)"))
+        .setType("number")
     );
 
     const sepiaEffect = extension
-      .addEffect('Sepia')
-      .setFullName(_('Sepia'))
-      .setDescription(_('Alter the colors to sepia'))
-      .addIncludeFile('Extensions/Effects/sepia-pixi-filter.js');
+      .addEffect("Sepia")
+      .setFullName(_("Sepia"))
+      .setDescription(_("Alter the colors to sepia"))
+      .addIncludeFile("Extensions/Effects/sepia-pixi-filter.js");
     const sepiaProperties = sepiaEffect.getProperties();
     sepiaProperties.set(
-      'opacity',
-      new gd.PropertyDescriptor(/* defaultValue= */ '1')
-        .setLabel(_('Opacity (between 0 and 1)'))
-        .setType('number')
+      "opacity",
+      new gd.PropertyDescriptor(/* defaultValue= */ "1")
+        .setLabel(_("Opacity (between 0 and 1)"))
+        .setType("number")
     );
 
-    return extension;
+    const displacementEffect = extension
+      .addEffect("Displacement")
+      .setFullName(_("Displacement"))
+      .setDescription(_("Add Displacement effect"))
+      .addIncludeFile("Extensions/Effects/displacement-pixi-filter.js");
+    const displacementProperties = displacementEffect.getProperties();
+    displacementProperties.set(
+      "displacementMapImage",
+      new gd.PropertyDescriptor("")
+        .setType("resource")
+        .addExtraInfo("image")
+        .setLabel(_("Displacement map image for the effect"))
+    );
+    displacementProperties.set(
+      "scaleX",
+      new gd.PropertyDescriptor(/* defaultValue=20 */ "20")
+        .setLabel(_("Scale X"))
+        .setType("number")
+    );
+    displacementProperties.set(
+      "scaleY",
+      new gd.PropertyDescriptor(/* defaultValue=20 */ "20")
```

Comment:
Write just /* defaultValue= */ "20"

The defaultValue is here to give an empty about what the "20" is :)